src/test/scala/viper/gobra/parsing/GoParserUnitTests.scala

```diff
@@ -0,0 +1,197 @@
+    "package p; func f() { var s []int; _ = s[i: /* ERROR \"2nd index required\" */ :k] };",
+    "package p; func f() { var s []int; _ = s[i: /* ERROR \"2nd index required\" */ :] };",
+    "package p; func f() { var s []int; _ = s[: /* ERROR \"2nd index required\" */ :] };",
+    "package p; func f() { var s []int; _ = s[: /* ERROR \"2nd index required\" */ ::] };",
+    "package p; func f() { var s []int; _ = s[i:j:k: /* ERROR \"expected ']'\" */ l] };",
+    "package p; func f() { for x /* ERROR \"boolean or range expression\" */ = []string {} }",
+    "package p; func f() { for x /* ERROR \"boolean or range expression\" */ := []string {} }",
+    "package p; func f() { for i /* ERROR \"boolean or range expression\" */ , x = []string {} }",
+    "package p; func f() { for i /* ERROR \"boolean or range expression\" */ , x := []string {} }",
+    "package p; func f() { go f /* ERROR HERE \"function must be invoked\" */ }",
+    "package p; func f() { defer func() {} /* ERROR HERE \"function must be invoked\" */ }",
+    "package p; func f() { go func() { func() { f(x func /* ERROR \"missing ','\" */ (){}) } } }",
+    "package p; func _() (type /* ERROR \"found 'type'\" */ T)(T)",
+    "package p; func (type /* ERROR \"found 'type'\" */ T)(T) _()",
+    "package p; type _[A+B, /* ERROR \"expected ']'\" */ ] int",
+
+    "package p; var a = a[[]int:[ /* ERROR \"expected expression\" */ ]int];",
+
+    // "package p; type I1 interface{}; type I2 interface{ (/* ERROR \"expected '}', found '('\" */ I1) }",
+
+    // issue 8656
+    "package p; func f() (a b string /* ERROR \"missing ','\" */ , ok bool)",
+
+    // issue 9639
+    "package p; var x /* ERROR \"missing variable type or initialization\" */ , y, z;",
+    "package p; const x /* ERROR \"missing constant value\" */ ;",
+    "package p; const x /* ERROR \"missing constant value\" */ int;",
+    "package p; const (x = 0; y; z /* ERROR \"missing constant value\" */ int);",
+
+    // issue 12437
+    "package p; var _ = struct { x int, /* ERROR \"expected ';', found ','\" */ }{};",
+    "package p; var _ = struct { x int, /* ERROR \"expected ';', found ','\" */ y float }{};",
+
+    // issue 11611
+    "package p; type _ struct { int, } /* ERROR \"expected 'IDENT', found '}'\" */ ;",
+    "package p; type _ struct { int, float } /* ERROR \"expected type, found '}'\" */ ;",
+
+    // issue 13475
+    "package p; func f() { if true {} else ; /* ERROR \"expected if statement or block\" */ }",
+    "package p; func f() { if true {} else defer /* ERROR \"expected if statement or block\" */ f() }",
+  )
+
+
+  for (source <- valids) yield {
+    test(source) {
+      frontend.parseProgram(source) should matchPattern {
+        case Right(_) =>
+      }
+    }
+  }
+
+  for (source <- invalids) yield {
+    test(source) {
+      frontend.parseProgram(source) should be
+    }
+  }
+
+
+
+  /* ** Stubs, mocks and other test setup */
+
+  class TestFrontend {
```

Comment:
this seems to be copy pasted from the existing parser unit tests. Please move them to a separate file such that both tests can make use of them without code duplication